From 03664a62914782dbd8f069bbcf8a0c8ca1df7010 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Wed, 20 Aug 2014 14:34:23 +0200
Subject: [PATCH 4/9] resolved-dns-rr: fix typo

a->rrsig.type_covered != a->rrsig.type_covered" is always false
regardless of the values of its operands because those operands are identical.
---
 src/resolve/resolved-dns-rr.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/resolve/resolved-dns-rr.c b/src/resolve/resolved-dns-rr.c
index c792ded..c5f7cb9 100644
--- a/src/resolve/resolved-dns-rr.c
+++ b/src/resolve/resolved-dns-rr.c
@@ -425,13 +425,13 @@ int dns_resource_record_equal(const DnsResourceRecord *a, const DnsResourceRecor
 
         case DNS_TYPE_RRSIG:
                 /* do the fast comparisons first */
-                if (a->rrsig.type_covered != a->rrsig.type_covered ||
-                    a->rrsig.algorithm != a->rrsig.algorithm ||
-                    a->rrsig.labels != a->rrsig.labels ||
-                    a->rrsig.original_ttl != a->rrsig.original_ttl ||
-                    a->rrsig.expiration != a->rrsig.expiration ||
-                    a->rrsig.inception != a->rrsig.inception ||
-                    a->rrsig.key_tag != a->rrsig.key_tag ||
+                if (a->rrsig.type_covered != b->rrsig.type_covered ||
+                    a->rrsig.algorithm != b->rrsig.algorithm ||
+                    a->rrsig.labels != b->rrsig.labels ||
+                    a->rrsig.original_ttl != b->rrsig.original_ttl ||
+                    a->rrsig.expiration != b->rrsig.expiration ||
+                    a->rrsig.inception != b->rrsig.inception ||
+                    a->rrsig.key_tag != b->rrsig.key_tag ||
                     a->rrsig.signature_size != b->rrsig.signature_size ||
                     memcmp(a->rrsig.signature, b->rrsig.signature, a->rrsig.signature_size) != 0)
                         return false;
-- 
2.1.0

